File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.java
class Solution {
    public int solution(String s) {
        int ans = s.length();
        for (int i = 1; i <= s.length() / 2 + 1; i++) {
            int idx = 0;
            StringBuilder sb = new StringBuilder();
            while (idx < s.length()) {
                int cnt = 0, j = idx;
                if(idx + i < s.length()){
                    String cur = s.substring(idx, idx + i);
                    while(j < s.length()){
                        String next = s.substring(j, Math.min(j+i, s.length()));
                        if (cur.equals(next)){
                            cnt ++;
                        }else{
                            break;
                        }
                        j += i;
                    }
                    if(cnt > 1){
                        sb.append(cnt);
                    }
                    sb.append(cur);
                    idx = j;
                }else{
                    sb.append(s.substring(idx));
                    break;
                }
            }
            ans = Math.min(ans, sb.toString().length());
        }
        return ans;
    }
}

File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.py
def solution(s):
    ans, s_len = len(s), len(s)
    for l in range(1, s_len // 2 + 1):
        idx, tmp = 0, []
        while idx + l <= s_len:
            cnt, cur = 1, s[idx:idx+l]
            while idx+l*2 <= s_len and cur == s[idx+l:idx+l*2]:
                idx += l
                cnt += 1
            idx += l
            
            if cnt > 1:
                tmp.append(cnt)
            
            tmp.append(cur)
            
        if idx < s_len:
            tmp.append(s[idx:])
            
        ans = min(ans, len("".join(map(str, tmp))))
            
    return ans